### PR TITLE
[TBBAS-1732] Python: Readers structs support

### PR DESCRIPTION
**Pull request by Yarikk26**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description:
Python readers output **Struct** typed samples as **NumPy Structured Arrays**

### CODE REVIEW

**Review by dejancrn — APPROVED** [review]
Looks ok, however, it would be nice to add example on how to read structs. An example should use reference device CAN signal as a source and a stream reader used to read the can messages.

**Comment by Yarikk26** [comment]
Added the reading CAN example

File: bindings/python/opendaq/generated/signal/py_mock_signal.cpp [hunk 1]
```diff
@@ -68,14 +67,148 @@ void MockSignal::addData(const py::array_t<double>& data)
...
     signal.sendPacket(valuePacket);
     domainSignal.sendPacket(domainPacket);
     offset += size;
 }
 
+void MockSignal::addObjects(const py::object& objects, bool updateDescriptor)
+{
+    daq::DataDescriptorPtr descriptor;
+    size_t size = 1;
+    if (py::isinstance<py::dict>(objects) || py::isinstance<py::list>(objects))
```
Comment (aljazfrancic) [inline bindings/python/opendaq/generated/signal/py_mock_signal.cpp:82]: Is outer `if` necessary?

File: bindings/python/opendaq/generated/signal/py_mock_signal.cpp [hunk 2]
```diff
@@ -68,14 +67,148 @@ void MockSignal::addData(const py::array_t<double>& data)
...
+}
+
+daq::DataDescriptorPtr MockSignal::createDataDescriptor(const py::object& object, const std::string& name)
+{
+    if (py::isinstance<py::list>(object))
+    {
+        const auto& list = py::cast<py::list>(object);
+        if (list.empty())
+        {
+            throw py::value_error("Cannot create DataDescroptor from empty list");
```
Comment (aljazfrancic) [inline bindings/python/opendaq/generated/signal/py_mock_signal.cpp:169]: `DataDescroptor` to `DataDescriptor` (typo).

**Review by aljazfrancic — APPROVED** [review]
Look good to me, except the minor details mentioned in the review.

File: bindings/python/opendaq/generated/signal/py_mock_signal.cpp (hunk 1 above)
Comment (Yarikk26) [reply bindings/python/opendaq/generated/signal/py_mock_signal.cpp:82]: Yep. Just to filter non dict or list values passed

**Pull request merged**